ipapython/certdb.py

```diff
@@ -722,8 +722,13 @@ def import_files(self, files, import_keys=False, key_password=None,
             if import_keys:
                 try:
                     self.import_pkcs12(filename, key_password)
-                except RuntimeError:
-                    pass
+                except RuntimeError as e:
+                    if "unknown error import pkcs#12 file" in str(e):
```

Comment:
@flo-renaud sorry to be a nit-pick, but I'm not comfortable with this "stringly-typed" approach.  Later when someone updates one string they will easily miss the other, and break the program.  We could create new exception subclasses to convey the errors from `import_pkcs12`, and use different `except` blocks to handle the different cases.

Apart from that the logic now seems fine to me, and I am glad you added comments about it :)

Reply:
Thanks for your comment. You are right, it is cleaner to raise subclasses of RuntimeError and will be easier to maintain. I just pushed a new version.